src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -779,9 +819,9 @@ internal async Task SendInvocationRequest(ScriptInvocationContext context)
                     InvocationRequest = invocationRequest
                 });
 
-                if (_cancelCapabilityEnabled != null && _cancelCapabilityEnabled.Value)
+                if (_isHandlesInvocationCancelMessageCapabilityEnabled)
                 {
-                    context.CancellationToken.Register(() => SendInvocationCancel(invocationRequest.InvocationId));
+                    _cancellationCtr = context.CancellationToken.Register(() => SendInvocationCancel(invocationRequest.InvocationId));
```

Comment:
This is variable is scoped to this type, but being overridden for every request. Are you trying to keep track of this registration so you can remove it? If so, you may need to do something like storing this is the context and retrieving it when the invocation completes.

Reply:
Yeah I think there was feedback somewhere from Jacob about needing to dispose of the registeration - will store this in context similar to how we handle this in dni